### ISSUE TITLE
Sync bootcamp application stage with Hubspot deal

### ISSUE THREAD

**Issue by mbertrand** [mitodl/bootcamp-ecommerce#542]
Depends on #545 

As a marketing team member, I need to see the current stage of a bootcamp applicant in hubspot, so I can nurture them through the application, payment and enrollment process. 

Update the 'application_stage' property of a Hubspot deal whenever the user's application state changes.

Formerly synced via smapply and the `PersonalPrice.application_stage` field:
https://github.com/mitodl/bootcamp-ecommerce/issues/304
https://github.com/mitodl/bootcamp-ecommerce/pull/308
https://github.com/mitodl/bootcamp-ecommerce/issues/291
https://github.com/mitodl/bootcamp-ecommerce/pull/299

**Comment by pdpinch** [comment]
There's a couple of stages missing from the current list. This should be complete. 


```
 STARTED = "APPLICATION STARTED"
 AWAITING_PROFILE_COMPLETION = "AWAITING_PROFILE_COMPLETION" 
 AWAITING_RESUME = "AWAITING_RESUME" 
 AWAITING_USER_SUBMISSIONS = "AWAITING_USER_SUBMISSIONS" 
 AWAITING_SUBMISSION_REVIEW = "AWAITING_SUBMISSION_REVIEW" 
 AWAITING_PAYMENT = "AWAITING_PAYMENT" 
 ENROLLED = "ENROLLED" 
 WAITLISTED = "WAITLISTED"
 REJECTED = "REJECTED" 
 DEFERRED = "DEFERRED" 
```

**Comment by Ferdi** [comment]
@pdpinch  vidoe interview is missing ?

**Comment by pdpinch** [comment]
Thanks @Ferdi I added it to the comment above, https://github.com/mitodl/bootcamp-ecommerce/issues/542#issuecomment-631011556

Let us know if it's not in the right spot, or if the stage name isn't clear enough.

**Comment by Ferdi** [comment]
@pdpinch great, thanks.  @gsidebo how do we do it ?

**Comment by odlbot** [comment]
➤ Matt Bertrand commented:

pdpinch ferdi regarding the additional application stages, seems like adding those should be a separate issue/PR?

How do you want them to show up in hubspot - in a more user-friendly version like "AWAITING_USER_SUBMISSIONS" => "Awaiting user submissions", or just sync them as is?

**Comment by gsidebo** [comment]
@mbertrand @Ferdi I think the more important question is how specific we need to be with the state. Is it sufficient to show "Awaiting user submissions", or does it need to be more specific like "Awaiting video interview"?

**Comment by gsidebo** [comment]
@mbertrand Just spoke w/ Ferdi. We do want to be specific about the type of user submission we're waiting on. That means either (a) sending "awaiting video interview" to hubspot instead of "awaiting user submissions", or (b) adding "video interview" to the data set that we send to hubspot

**Comment by mbertrand** [comment]
@gsidebo will those be separate states?  The code I'm working on simply sends the `application.state` value to hubspot.  If the state is `AWAITING_VIDEO_INTERVIEW`, then that's what will show up there as well.

**Comment by gsidebo** [comment]
@mbertrand `AWAITING_VIDEO_INTERVIEW` isn't a state in the app. The valid states are defined in `applications/constants.py`. Your signal can have some logic to change the text or add fields to the data we send to hubspot if the actual `state` value is `AWAITING_USER_SUBMISSIONS`

**Comment by mbertrand** [comment]
@gsidebo I was wondering if `constants.py` needed to be updated to have additional values like `AWAITING_VIDEO_INTERVIEW`.  If that isn't going to change, then it seems the code needs to look up whether the application has a video submission step or a quiz submission step.  What if an application requires both and the user has submitted neither?  Should the hubspot application stage be "Awaiting Video Interview, Awaiting Quiz"?

**Comment by mbertrand** [comment]
@pdpinch @ferdi it looks to me that if the state is in `AWAITING_USER_SUBMISSIONS`, the user might need to submit a quiz or a video in the next application step.  If the user needs to submit a quiz, do you want an `AWAITING_QUIZ_SUBMISSION` status?

**Comment by gsidebo** [comment]
@mbertrand The order of the submissions for a bootcamp is defined in `ApplicationStep` via the `step_order` field. If the application requires both, we can inspect the application steps defined and compare them to the user's submissions to figure out which one is needed next
